tasks/src/main/scala/dagr/tasks/picard/CollectIlluminaBasecallingMetrics.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package dagr.tasks.picard
+
+import dagr.tasks.DagrDef.{DirPath, FilePath}
+
+import scala.collection.mutable.ListBuffer
+
+object CollectIlluminaBasecallingMetrics {
+  def toMetricsFileName(lane: Int): String = s".lane-$lane.metrics"
+}
+
+class CollectIlluminaBasecallingMetrics(basecallsDir: DirPath,
+                                        lane: Int,
+                                        barcodeFile: Option[FilePath] = None,
+                                        readStructure: String,
+                                        outputDir: Option[DirPath] = None,
+                                        barcodesDir: Option[DirPath] = None
+                                       ) extends PicardTask with PicardMetricsTask {
+
+  override def in: FilePath = outputDir.getOrElse(basecallsDir).resolve("basecalling")
```

Comment:
This is bizarre.  Why would we prefer the output directory as the input directory?